sql/sql_parse.cc

```diff
@@ -4117,7 +4117,15 @@ int mysql_execute_command(THD *thd, bool first_level) {
         if (!lex->drop_temporary &&
             (!thd->is_current_stmt_binlog_format_row() ||
              !find_temporary_table(thd, table))) {
-          WSREP_TO_ISOLATION_BEGIN(NULL, NULL, all_tables);
+          // We cannot use WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF, because here
+          // lex->no_write_to_binlog is uninitialized
+          wsrep::key_array keys;
+          wsrep_append_fk_parent_table(thd, all_tables, &keys);
+          if (WSREP(thd) &&
+              wsrep_to_isolation_begin(thd, NULL, NULL, all_tables, NULL, NULL,
+                                       &keys)) {
```

Comment:
Just a thought. Shouldn't we also check for the below condition?
```
thd->wsrep_cs().state() != wsrep::client_state::s_none
```
If we were using `WSREP_TO_ISOLATION_BEGIN_FK_TABLES_IF`, then we would have done the above check. But it looks we don't do it now. 

Can you check this?

Reply:
As we discussed on Slack, as it was not checked before as well, let's leave it for now as it is.

Reply:
Can you guys please point to me to slack conversation on dev-server?

Reply:
The conversation happened over IM and the summary of was that, Kamil had indeed done some test with the condition we had proposed, and saw no difference in behavior. So, it was hard to judge the correct fix.

Also, considering the fact that the previous condition didn't have the check for the wsrep::client_state, we found it to be safer to revert to previous implementation and not push an unsure patch.

Reply:
Thanks for the explanation!